client/src/boot/index.js

```diff
@@ -1,134 +1,82 @@
+  ReducerRegister.add('apollo', apolloClient.reducer());
+  ReducerRegister.add('treeDropdownField', TreeDropdownFieldReducer);
   // @todo - Restore this once we address https://github.com/silverstripe/silverstripe-admin/issues/90
-  // reducerRegister.add('unsavedForms', UnsavedFormsReducer);
-
-  // Force this to the end of the execution queue to ensure it's last.
-  window.setTimeout(() => {
-    registerComponents();
-    Injector.load();
-  }, 0);
+  // ReducerRegister.add('unsavedForms', UnsavedFormsReducer);
 
-  const initialState = {};
-  const rootReducer = combineReducers(reducerRegister.getAll());
+  const rootReducer = combineReducers(ReducerRegister.getAll());
   const middleware = [
     thunkMiddleware,
     apolloClient.middleware(),
   ];
 
-  const env = Config.get('environment');
   const debugging = Config.get('debugging');
   let runMiddleware = applyMiddleware(...middleware);
 
-  // use browser extension `compose` function if it's available
-  const composeExtension = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__;
-  // use browser extension devTools if it's available
-  // this is the old way: `devToolsExtension` is being deprecated
-  const devTools = window.__REDUX_DEVTOOLS_EXTENSION__ || window.devToolsExtension;
-
-  if (env === 'dev' && debugging) {
-    if (typeof composeExtension === 'function') {
-      // use compose from extension first
-      runMiddleware = composeExtension(applyMiddleware(...middleware));
-    } else if (typeof devTools === 'function') {
-      // fallback to old way
-      runMiddleware = compose(applyMiddleware(...middleware), devTools());
-    }
+  if (process.env.NODE_ENV !== 'production' && debugging) {
+    runMiddleware = applyDevtools(runMiddleware);
   }
 
   const createStoreWithMiddleware = runMiddleware(createStore);
-  const store = createStoreWithMiddleware(rootReducer, initialState);
+  const store = createStoreWithMiddleware(rootReducer, {});
 
   // Set the initial config state.
-  store.dispatch(configActions.setConfig(Config.getAll()));
+  store.dispatch(setConfig(Config.getAll()));
 
-  // Expose store for legacy use
-  window.ss = window.ss || {};
+  // Expose store stuff for legacy use
   window.ss.store = store;
-
-  // Expose client for legacy use
-  window.ss = window.ss || {};
   window.ss.apolloClient = apolloClient;
 
-  // Bootstrap routing
-  const routes = new BootRoutes(store, apolloClient);
-  routes.start(window.location.pathname);
-
-  // @TODO - Remove once we remove entwine
+  // @todo - Remove once we remove entwine
   // Enable top-level css selectors for react-dependant entwine sections
   if (window.jQuery) {
     window.jQuery('body').addClass('js-react-boot');
   }
+
+  // Bootstrap routing
+  const routes = new BootRoutes(store, apolloClient);
+
+  // Force this to the end of the execution queue to ensure it's last.
+  window.setTimeout(() => {
+    registerComponents();
+    Injector.load();
+    routes.start(window.location.pathname);
```

Comment:
Yeah, that's cool
I've always felt uneasy with using `window.onload` for `appBoot` in general, but it seems unavoidable currently.